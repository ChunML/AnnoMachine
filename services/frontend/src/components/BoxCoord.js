import React from 'react';
import PropTypes from 'prop-types';

function BoxCoord({ coords }) {
  return (
    <React.Fragment>
      <span className="box-coord">{Math.floor(coords.x_min)}</span>
      <span className="box-coord">{Math.floor(coords.y_min)}</span>
      <span className="box-coord">{Math.floor(coords.x_max)}</span>
      <span className="box-coord">{Math.floor(coords.y_max)}</span>
    </React.Fragment>
  );
}

BoxCoord.propTypes = {
  coords: PropTypes.object.isRequired,
};

export default BoxCoord;
