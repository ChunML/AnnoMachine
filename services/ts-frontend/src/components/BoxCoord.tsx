import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { BoxType } from '../App';

function BoxCoord({ coords }: { coords: BoxType }) {
  const [box, setBox] = useState<BoxType>(coords);

  useEffect(() => {
    setBox(coords);
  }, [coords]);

  return (
    <React.Fragment>
      <span className="box-coord">{Math.floor(box.x_min)}</span>
      <span className="box-coord">{Math.floor(box.y_min)}</span>
      <span className="box-coord">{Math.floor(box.x_max)}</span>
      <span className="box-coord">{Math.floor(box.y_max)}</span>
    </React.Fragment>
  );
}

BoxCoord.propTypes = {
  coords: PropTypes.object.isRequired,
};

export default BoxCoord;
