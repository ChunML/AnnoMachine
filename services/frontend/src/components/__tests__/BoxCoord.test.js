import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import BoxCoord from '../BoxCoord';
import '../../setupTests';

const coords = {
  x_min: 10,
  y_min: 10,
  x_max: 20,
  y_max: 20,
};

it('BoxCoord renders properly', () => {
  const wrapper = shallow(<BoxCoord coords={coords} />);
  const spans = wrapper.find('span');
  expect(spans.length).toBe(4);
});

it('BoxCoord renders a snapshot properly', () => {
  const tree = renderer.create(<BoxCoord coords={coords} />).toJSON();
  expect(tree).toMatchSnapshot();
});
