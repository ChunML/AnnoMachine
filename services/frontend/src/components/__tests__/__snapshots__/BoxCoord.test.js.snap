// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxCoord renders a snapshot properly 1`] = `
Array [
  <span
    style={
      Object {
        "background": "powderblue",
        "borderRadius": "4px",
        "color": "maroon",
        "letterSpacing": "0.2rem",
        "margin": "5px",
        "padding": "5px 10px",
      }
    }
  >
    10
  </span>,
  <span
    style={
      Object {
        "background": "powderblue",
        "borderRadius": "4px",
        "color": "maroon",
        "letterSpacing": "0.2rem",
        "margin": "5px",
        "padding": "5px 10px",
      }
    }
  >
    10
  </span>,
  <span
    style={
      Object {
        "background": "powderblue",
        "borderRadius": "4px",
        "color": "maroon",
        "letterSpacing": "0.2rem",
        "margin": "5px",
        "padding": "5px 10px",
      }
    }
  >
    20
  </span>,
  <span
    style={
      Object {
        "background": "powderblue",
        "borderRadius": "4px",
        "color": "maroon",
        "letterSpacing": "0.2rem",
        "margin": "5px",
        "padding": "5px 10px",
      }
    }
  >
    20
  </span>,
]
`;
